script/deploy-dev.mk

```diff
@@ -0,0 +1,52 @@
+# Deploy a dev environment of Kubeapps using OIDC for authentication with a
+# local dex as the provider.
+#
+# Targets in this helper assume that kubectl is configured with a cluster
+# that has been setup with OIDC support (see ./cluster-kind.mk)
+
+deploy-helm:
+	kubectl apply -f ./docs/user/manifests/kubeapps-local-dev-tiller-rbac.yaml
+	helm init --service-account tiller --wait
+
+deploy-dex: deploy-helm
+	kubectl create namespace dex
+	kubectl -n dex create secret tls dex-web-server-tls \
+		--key ./script/test-certs/dex.key.pem \
+		--cert ./script/test-certs/dex.cert.pem
+	helm install stable/dex --namespace dex --name dex --version 2.4.0 \
+		--values ./docs/user/manifests/kubeapps-local-dev-dex-values.yaml
+
+# The api server does not have service dns entries (in kind or vanilla k8s), so
+# dex is normally required to be available on an external host. Short-circuit
+# that requirement by ensuring dex.dex resolves to the internal IP address on
+# the apiserver so that it can initialise the oidc plugin.
+update-apiserver-etc-hosts:
+	while ! kubectl -n kube-system get po kube-apiserver-kubeapps-control-plane; do \
+		echo "Waiting for api server" && sleep 1; \
+	done
+	kubectl -n kube-system exec kube-apiserver-kubeapps-control-plane -- \
```

Comment:
I am curious, it doesn't work if you add an ingress to dex, right? The K8s API can't resolve ingress URLs?

Reply:
The resolv.conf used on the api-server pod doesn't have any internal nameserver - see https://github.com/kubernetes-sigs/kind/issues/1077 where I dumped /etc/resolv.conf etc. So if I used an actual k8s cluster, I could create an ingress with an externally resolvable hostname and it would work, but I can't see how that can be used for a local setup?

What I was able to do is set the issuer-url for the apiserver to `https://localhost/api/v1/ser..../proxy` which worked fine for the api server, but the issuer-url needs to match the issuer url specified by dex, which cannot be reconciled because the api server proxy url proxies through to the root of the dex service... if dex is configured with the issuerurl as https://localhost/ the apiserver fails because the issuer urls don't match, and if you make the dex one match, then the proxied request 404s (as it's hitting the root /.well-known/..., while dex is serving from /api/v1/.../proxy).

Reply:
I see, just wondering if I made this work with Minikube changing the `/etc/hosts` or with an ingress (probably the former but I don't remember). Thanks for the info!